### ISSUE TITLE
[lines-between-class-members] does not make sense with overloading

### ISSUE THREAD

**Issue by emilekberg** [typescript-eslint/typescript-eslint#1610]
I understand this rule for most scenarios, but for this scenario i don't really see what makes it easier to read. the extra lines, to me, makes it more confusing. 
**What code were you trying to parse?**

```ts
  with(name: string, type: FieldTypes): SchemaBuilder
  with(name: string, type: FieldTypes.utf8, maxStringSize: number): SchemaBuilder
  with(name: string, type: FieldTypes, size?: number): SchemaBuilder {
    this.fields.push({
      name,
      type,
      size: getSize(type, size),
    });
    return this;
  }
```

**What did you expect to happen?**
No warning, it does not make sense here
**What actually happened?**
  43:3   error  Expected blank line between class members  lines-between-class-members
  44:3   error  Expected blank line between class members  lines-between-class-members
**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.2` |
| `@typescript-eslint/parser`               | `2.19.2` |
| `TypeScript`                              | `3.7.5` |
| `ESLint`                                  | `6.8.0` |
| `node`                                    | `13.6.0` |
| `npm`                                     | `6.13.6` |

**Comment by bradzacher** [comment]
happy to accept a PR if you want to create an extension of the rule that ignores overloads.
